### PR TITLE
Compilation fixes for Noetic

### PR DESCRIPTION
**Pull request by wxmerkt**
If Anymal were to be used, it'd be important to check which Melodic release the laptops are on: Anybotics' 20.05 and 20.07.2 are highly likely to require the fixes for `position_controller_cdt` as well since the rocoma stack was deprecated and replaced with the motion control stack, unless launched in `legacy` mode.

### CODE REVIEW

**Comment by mauricefallon** [comment]
We wont use ANYmal for the CDT, but its useful to keep this code alive.
As long as it compiles that is sufficient for the melodic to noetic transition

**Comment by benoit-robotics** [comment]
@wxmerkt This branch doesn't build on melodic. Does it need to ?

File: lidar_navigation_cdt/include/grid_map_cdt/Challenge.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 #include <grid_map_ros/grid_map_ros.hpp>
 
-#include <filters/filter_chain.h>
+#include <filters/filter_chain.hpp>
```
Comment (wxmerkt) [inline lidar_navigation_cdt/include/grid_map_cdt/Challenge.hpp]: It doesn't build due to this - we need to change back to `.h` for it to compile on both Melodic and Noetic

**Comment by wxmerkt** [comment]
@benoit-robotics I just added a note which line needs to change to be able to build on both OS versions

File: lidar_navigation_cdt/include/grid_map_cdt/Challenge.hpp (hunk 1 above)
Comment (benoit-robotics) [reply lidar_navigation_cdt/include/grid_map_cdt/Challenge.hpp]: there is the same issue in navigation_drs

Comment (benoit-robotics) [reply lidar_navigation_cdt/include/grid_map_cdt/Challenge.hpp]: I will fix it

Comment (wxmerkt) [reply lidar_navigation_cdt/include/grid_map_cdt/Challenge.hpp:17]: Would you like me to do the changes in both or are you handling it?

**Pull request merged**